pandas/core/indexes/base.py

```diff
@@ -501,6 +500,18 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
     See each method's docstring.
     """
 
+    @property
+    def asi8(self):
+        """
+        Integer representation of the values.
```

Comment:
Only for the datetime-like indexes (we should probably rather remove it, but that's another issue)